www-apps/redmine/redmine-3.3.3.ebuild

```diff
@@ -0,0 +1,211 @@
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+USE_RUBY="ruby21 ruby22"
+inherit eutils depend.apache ruby-ng user
+
+DESCRIPTION="Flexible project management web application using the Ruby on Rails framework"
+HOMEPAGE="http://www.redmine.org/"
+SRC_URI="http://www.redmine.org/releases/${P}.tar.gz"
+
+KEYWORDS="~amd64"
+LICENSE="GPL-2"
+SLOT="0"
+# All db-related USEs are ineffective since we depend on rails
+# which depends on activerecord which depends on all ruby's db bindings
+#IUSE="ldap openid imagemagick postgres sqlite mysql fastcgi passenger"
+IUSE="imagemagick fastcgi ldap markdown passenger"
+
+# nokogiri ruby 2.1 >= 1.7.0, else 1.6.8
+
+ruby_add_rdepend "
+	dev-ruby/actionpack-action_caching
+	dev-ruby/actionpack-xml_parser:*
+	>=dev-ruby/builder-3.2.2:3.2
+	>=dev-ruby/coderay-1.1.0
+	dev-ruby/i18n:0.7
+	>=dev-ruby/rails-html-sanitizer-1.0.3
+	>=dev-ruby/rack-openid-0.2.1
+	>=dev-ruby/jquery-rails-3.1.4:3
+	dev-ruby/mime-types:*
+	dev-ruby/mimemagic
+	>=dev-ruby/nokogiri-1.6.8
+	dev-ruby/protected_attributes
+	>=dev-ruby/rails-4.2.5.2:4.2
+	dev-ruby/request_store:1.0.5
+	>=dev-ruby/roadie-rails-1.1.0
+	>=dev-ruby/rbpdf-1.19.2
+	>=dev-ruby/ruby-openid-2.3.0
+	dev-ruby/rubygems
+	fastcgi? ( dev-ruby/fcgi )
+	imagemagick? ( >=dev-ruby/rmagick-2.14.0 )
+	ldap? ( >=dev-ruby/ruby-net-ldap-0.12.0 )
+	markdown? ( >=dev-ruby/redcarpet-3.3.2 )
+	passenger? ( www-apache/passenger )
+	"
+
+#ruby_add_bdepend ">=dev-ruby/rdoc-2.4.2
+#	dev-ruby/yard
+#	test? (
+#		>=dev-ruby/shoulda-3.3.2
+#		>=dev-ruby/mocha-0.13.3
+#		>=dev-ruby/capybara-2.0.0
+#		<dev-ruby/nokogiri-1.6.0
+#	)"
+
+REDMINE_DIR="/var/lib/${PN}"
+
+pkg_setup() {
+	enewgroup redmine
+	enewuser redmine -1 -1 "${REDMINE_DIR}" redmine
+}
+
+all_ruby_prepare() {
+	rm -r log files/delete.me || die
+
+	# bug #406605
+	rm .gitignore .hgignore || die
+
+	rm Gemfile || die
+
+	echo "CONFIG_PROTECT=\"${EPREFIX}${REDMINE_DIR}/config\"" > "${T}/50${PN}"
+	echo "CONFIG_PROTECT_MASK=\"${EPREFIX}${REDMINE_DIR}/config/locales ${EPREFIX}${REDMINE_DIR}/config/settings.yml\"" >> "${T}/50${PN}"
+
+	# remove ldap staff module if disabled to avoid #413779
+	use ldap || rm app/models/auth_source_ldap.rb || die
+
+	# Make it work
+	sed -i -e "1irequire 'request_store'" app/controllers/application_controller.rb || die
+	sed -i -e "18irequire 'action_controller'" -e "19irequire 'action_controller/action_caching'"\
+		app/controllers/welcome_controller.rb || die
+	sed -i -e "4irequire 'action_dispatch/xml_params_parser'" -e "/Bundler/d" config/application.rb || die
+	sed -i -e "18irequire 'protected_attributes'" app/models/custom_field.rb || die
+	sed -i -e "19irequire 'roadie-rails'" app/models/mailer.rb || die
+}
+
+all_ruby_install() {
+	dodoc doc/{CHANGELOG,INSTALL,README_FOR_APP,RUNNING_TESTS,UPGRADING}
+	rm -r doc || die
+	dodoc README.rdoc
+	rm README.rdoc || die
+
+	keepdir /var/log/${PN}
+	dosym /var/log/${PN}/ "${REDMINE_DIR}/log"
```

Comment:
Please use a relative symlink here. Since REDMINE_DIR is unconditionally defined in the ebuild, you can do that safely.